### PR TITLE
create rabbit consumer etc

### PR DESCRIPTION
**Pull request by neildsouth**
### Description

adds rabbit consumer for canceled events (timed out)
adds a status and removes the unused `Ready` from ClinicalReviewRecord
update GetAllAsync to only return status of awaitingReview
update AcknowledgeClinicalReview to set status

A few sentences describing the changes proposed in this pull request.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] All tests passed locally.

### CODE REVIEW

File: src/Aide.ClinicalReview.Database/Interfaces/IClinicalReviewRepository.cs [hunk 1]
```diff
@@ -48,5 +48,12 @@ public interface IClinicalReviewRepository
         /// <param name="executionId">executionId of the task.</param>
         /// <param name="workflow">Workflow object to create.</param>
         Task<string> AcknowledgeAsync(string executionId, AcknowledgeClinicalReview acknowledge);
+
+        /// <summary>
+        /// Set review as canceled
+        /// </summary>
+        /// <param name="executionId">executionId of the task.</param>
+        /// <returns>Review Id</returns>
+        Task<string> CanceledAsync(string executionId);
```
Comment (mehranrashid) [inline src/Aide.ClinicalReview.Database/Interfaces/IClinicalReviewRepository.cs:57]: `Cancelled`

**Review by mehranrashid — APPROVED** [review]

File: src/Aide.ClinicalReview.Service/Handlers/AideClinicalReviewRequestCallBackHandler.cs [hunk 2]
```diff
@@ -158,7 +174,7 @@ private async Task ProcessSeriesRelativePaths(List<File> seriesRelativePaths, Cl
             {
                 var dcmFilesInfo = await _dicomService.GetAllDicomFileInfoInPath(seriesRelativePath.RelativeRootPath, seriesRelativePath.Bucket);
 
-                if (dcmFilesInfo.IsNullOrEmpty())
+                if (dcmFilesInfo is null || dcmFilesInfo.IsNullOrEmpty())
```
Comment (samrooke) [inline src/Aide.ClinicalReview.Service/Handlers/AideClinicalReviewRequestCallBackHandler.cs:177]: Doesn't the following cover both cases here?
`dcmFilesInfo.IsNullOrEmpty()`

**Review by samrooke — APPROVED** [review]

File: src/Aide.ClinicalReview.Service/Handlers/AideClinicalReviewRequestCallBackHandler.cs (hunk 2 above)
Comment (neildsouth) [reply src/Aide.ClinicalReview.Service/Handlers/AideClinicalReviewRequestCallBackHandler.cs:177]: it does, however the complier wasn't happy and the squiggly line annoyed me !

Comment (samrooke) [reply src/Aide.ClinicalReview.Service/Handlers/AideClinicalReviewRequestCallBackHandler.cs:177]: Fair enough 🙂

**Pull request merged**